### PR TITLE
Improved GHA CI build and local dev DX

### PR DESCRIPTION
**Pull request by victorskl**
* Bumped localstack v2
* Added reusable GHA container image caches.yml workflow
* Added mocking section in README

### CODE REVIEW

File: .github/workflows/caches.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  triage:
+    runs-on: ubuntu-latest
+    steps:
+      # Localstack
+      - name: Restore Localstack Image Cache if it exists
+        id: cache-docker-localstack
+        uses: actions/cache@v3
+        with:
+          path: ci/cache/docker/localstack
+          key: cache-docker-localstack-2.1
```
Comment (andrewpatto) [inline .github/workflows/caches.yml:23]: can you drag this "2.1" somehow up into a const or env variable.. its just that it then is used in about 10 different spots in the next 10 lines - easy for someone to miss changing it when bumping the version..

File: .github/workflows/caches.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        if: steps.cache-docker-localstack.outputs.cache-hit == 'true'
+        run: docker image load --input ./ci/cache/docker/localstack/localstack-2.1.tar
+
+      # MySQL
+      - name: Restore MySQL Image Cache if it exists
+        id: cache-docker-mysql
+        uses: actions/cache@v3
+        with:
+          path: ci/cache/docker/mysql
+          key: cache-docker-mysql-8
```
Comment (andrewpatto) [inline .github/workflows/caches.yml:39]: same for the myself 'version'

File: .github/workflows/caches.yml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+          path: ci/cache/docker/haproxy
+          key: cache-docker-haproxy-2.4
+
+      - name: Update HAProxy Image Cache if cache miss
+        if: steps.cache-docker-haproxy.outputs.cache-hit != 'true'
+        run: docker pull public.ecr.aws/docker/library/haproxy:2.4 && mkdir -p ci/cache/docker/haproxy && docker image save public.ecr.aws/docker/library/haproxy:2.4 --output ./ci/cache/docker/haproxy/haproxy-2.4.tar
+
+      - name: Use HAProxy Image Cache if cache hit
+        if: steps.cache-docker-haproxy.outputs.cache-hit == 'true'
+        run: docker image load --input ./ci/cache/docker/haproxy/haproxy-2.4.tar
```
Comment (andrewpatto) [inline .github/workflows/caches.yml:63]: in fact - is it possible to refactor this into a repeatably useable snippet (I believe github actions allows 'functions' of some sort)

**Review by brainstorm — APPROVED** [review]

File: .github/workflows/caches.yml (hunk 3 above)
Comment (andrewpatto) [reply .github/workflows/caches.yml:63]: oh - I see the whole thing is already a re-usable script.. all good

**Review by reisingerf — APPROVED** [review]

**Review by williamputraintan — APPROVED** [review]

File: .github/workflows/caches.yml (hunk 3 above)
Comment (victorskl) [reply .github/workflows/caches.yml:63]: Ah yes. But you are also right. Let me experiment with composite action in another feat branch...

Comment (victorskl) [reply .github/workflows/caches.yml]: FYI. Corresponding build is here and pass.
https://github.com/umccr/orcabus/actions/runs/5333413263/jobs/9663842683?pr=32

Comment (victorskl) [reply .github/workflows/caches.yml]: Gentle ping @andrewpatto

**Review by andrewpatto — APPROVED** [review]

**Pull request merged**